torch/_refs/nn/functional/__init__.py

```diff
@@ -445,7 +446,7 @@ def softplus(
     return torch.where(scaled_input > threshold, a, rhs)
 
 
-@register_decomposition(aten.hardshrink)
+@aten.hardshrink.default.py_impl(DispatchKey.Autograd)
```

Comment:
Does this have any implications for `torch.autograd.inference_mode`, where I believe autograd dispatch is disabled entirely? Maybe keep the decomposition just in case?

Reply:
@albanD might know this one?

Reply:
But I think it should be alright. This just says something along the lines of "register this a composite implicit when it comes to autograd, even if the original function was not registered as composite implicit"

Reply:
If you want composite implicit: `CompositeImplit` should be used here I think.

Reply:
Actually `CompositeImplicitAutograd` looks to have higher priority than the `Autograd` key so why not just register that?

Reply:
If I set this to `CompositeImplicitAutograd` and I try to compile  `    return F.hardshrink(x).backward(torch.ones_like(x))`, the dispatchers doesn't know how to decompose the backward. I remember @ezyang once explained to me why `Autograd` is the key to use here, but I can't remember...